packages/protocol/contracts/governance/Governance.sol

```diff
@@ -657,17 +727,36 @@ contract Governance is
     uint256[] calldata dataLengths,
     bytes32 salt
   ) external {
-    bytes32 hash = keccak256(abi.encode(values, destinations, data, dataLengths, salt));
+    if (isL2()) {
+      bytes32 hash = keccak256(abi.encode(values, destinations, data, dataLengths, salt));
+
+      (
+        bool approved,
+        bool councilApproved,
+        bool executed,
+        uint256 executionTimeLimit
+      ) = getL2HotfixRecord(hash);
+      require(!executed, "hotfix already executed");
+      require(approved, "hotfix not approved");
+      require(councilApproved, "hotfix not approved by security council");
+      require(executionTimeLimit >= now, "Execution time limit has already been reached.");
```

Comment:
Shouldn't this be rather executionTimeLimit < now since when executionTimeLimit >= now it is considered as timed out and can be reset ?

Reply:
`executionTimeLimit >= now` is still within the time `hotfixExecutionTimeWindow`. To reset the hotfix it needs to wait the the execution window passed. 

```require(hotfixes[hash].executionTimeLimit < now, "hotfix execution time limit not reached");```

check out the `GovernanceTest_resetHotfix` for the intended execution flow.